AmplifyPlugins/Auth/AWSAuthPlugin/Operations/AWSAuthSignUpOperation.swift

```diff
@@ -34,6 +34,12 @@ AuthSignUpOperation {
             return
         }
 
+        if let validationError = request.validate() {
```

Comment:
I think I've mentioned this before, but it's really confusing that `validate` returns an error, and that represents a failure case. We should either throw, so that `validate` has to handle it in a try/catch, or rename the method to `hasErrors` or similar.

Reply:
throwing error require to cast the resulting error to AmplifyAuthError again. So, making the method to "hasError" works better.